src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1323,17 +1324,17 @@ private static bool SupportsSpellCheck(JToken token)
         if (request.ProjectedKind == RazorLanguageKind.Html)
         {
             (synchronized, virtualDocumentSnapshot) = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync<HtmlVirtualDocumentSnapshot>(
-                request.HostDocument.Version,
-                request.HostDocument.Uri,
+                request.Identifier.Version,
+                request.Identifier.TextDocumentIdentifier.Uri,
                 rejectOnNewerParallelRequest: false,
                 cancellationToken);
             languageServerName = RazorLSPConstants.HtmlLanguageServerName;
         }
         else if (request.ProjectedKind == RazorLanguageKind.CSharp)
         {
             (synchronized, virtualDocumentSnapshot) = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync<CSharpVirtualDocumentSnapshot>(
-                request.HostDocument.Version,
-                request.HostDocument.Uri,
+                request.Identifier.Version,
+                request.Identifier.TextDocumentIdentifier.Uri,
                 rejectOnNewerParallelRequest: false,
```

Comment:
Just got to the end of this file, and I _love_ that all of the delegate requests don't construct new POCOs. Exactly what I was hoping was possible. Good work!